src/demo.rs

```diff
@@ -0,0 +1,111 @@
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::Path;
+use std::path::PathBuf;
+use std::thread;
+use tokio::fs::metadata;
+use tokio::fs::File;
+use tokio::io::{AsyncReadExt, AsyncSeekExt, SeekFrom};
+use tokio::sync::mpsc;
+use tokio::time::interval;
+use tokio::time::Duration;
+use serde::{Serialize, Deserialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BigBrotherPacket {
+    players: Vec<String>, // List of players on the server, array of SteamIDs
+    ip: String, // IP address of the server
+    sequence: i32, // A sequence number that starts at 0 and increments for each successive fragment
+    demo: Option<Vec<u8>>, // Up-to-date copy of demo file. None for packet 0 as the demo wouldn't be flushed to disk yet.
+}
+
+
+pub async fn demo_loop(demo_path: PathBuf) -> anyhow::Result<()> {
+    let (sync_tx, sync_rx) = std::sync::mpsc::channel::<Event>();
+    let (tx, mut rx) = mpsc::channel(32);
+    let config = Config::default().with_poll_interval(Duration::from_secs(2));
+
+    let mut watcher: RecommendedWatcher = Watcher::new(
+        Box::new(move |res: Result<Event, notify::Error>| {
+            let _ = sync_tx.send(res.unwrap());
+        }),
+        config,
+    )?;
+
+    watcher.watch(demo_path.as_path(), RecursiveMode::NonRecursive)?;
+
+    thread::spawn(move || loop {
+        if let Ok(event) = sync_rx.recv() {
+            let _ = tx.blocking_send(event); // This blocks in a separate thread so idgaf anymore
+        }
+    });
```

Comment:
Is there a point to having this separate thread receive events and then just immediately send them off again? Why not just send them straight to the final destination from the closure you passed to the watcher earlier?

Reply:
Tokio and notify do not play very nicely with each other, and this was one of the solutions I found that ended up working. This was also partially why I tried watchman. I'm hoping that I can iron this out and get it working within the tokio task.

Reply:
Ah I see, yes hopefully a way around this can be found.

Reply:
Nevermind I found a solution on reddit LOL